### PR TITLE
[WIP][Codegen] Codegen v2 kotlin

### PR DESCRIPTION
**Pull request by lamafab**
_NOTE: This is WIP and builds on top of https://github.com/trustwallet/wallet-core/pull/3122._

Current progress:

- [x] _(locally) Setup Kotlin Multiplatform env._
- [x] Remove unnecessary data from manifest.
- [x] Add all comments/docs from `include/` to manifest.
- [ ] Create templates and generate bindings for each platform:
	- [x] androidMain
		- [ ] C++ generated
	- [x] commonMain
	- [ ] ioMain
		- [x] _(2023-05-23) Started working on this_
	- [ ] jsMain
- [ ] Test all generated bindings, handle edge-cases.

### CODE REVIEW

File: codegen-v2/src/codegen/kotlin/mod.rs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+/// Convert the `TypeVariant` into the appropriate Swift type.
+impl From<TypeVariant> for KotlinType {
+    fn from(value: TypeVariant) -> Self {
+        let res = match value {
+            TypeVariant::Void => "Void".to_string(),
```
Comment (MaximPestryakov) [inline codegen-v2/src/codegen/kotlin/mod.rs:30]:
```suggestion
            TypeVariant::Void => "Unit".to_string(),
```

Comment (MaximPestryakov) [inline codegen-v2/src/codegen/kotlin/mod.rs:30]:
```suggestion
            TypeVariant::Void => "Unit".to_string(),
```

**Review by TassyBee — COMMENTED** [review]
Nice
